src/optimizer/projection_push_down_optimizer.cpp

```diff
@@ -185,7 +185,9 @@ void ProjectionPushDownOptimizer::visitDelete(LogicalOperator* op) {
             auto& rel = info.pattern->constCast<RelExpression>();
             collectExpressionsInUse(rel.getSrcNode()->getInternalID());
             collectExpressionsInUse(rel.getDstNode()->getInternalID());
-            collectExpressionsInUse(rel.getInternalIDProperty());
+            if (!rel.isEmpty() && rel.getRelType() == QueryRelType::NON_RECURSIVE) {
```

Comment:
So you don't need relID to delete?

First I think `rel.getRelType() == QueryRelType::NON_RECURSIVE` this should be assertion.

Second, `!rel.isEmpty()` this looks like a bug we didn't capture before. Is it true?

Reply:
> So you don't need relID to delete?

I do need. but not need it when rel is empty (no matched tables).

Changed `rel.getRelType() == QueryRelType::NON_RECURSIVE` to assertion.

> Second, !rel.isEmpty() this looks like a bug we didn't capture before. Is it true?

Not sure if it was a bug, but it gets exposed now when we don't always add rel ID property.